services-camel/src/main/java/edu/unc/lib/dl/services/camel/solr/CdrEventToSolrUpdateProcessor.java

```diff
@@ -87,24 +86,38 @@ public void process(Exchange exchange) throws Exception {
             childPids.add(PIDs.get(subject));
         }
 
-        if (MOVE.equals(solrActionType)) {
-            messageSender.sendIndexingOperation(userid, PIDs.get(targetId), childPids,
-                    IndexingActionType.MOVE);
-        } else if (ADD.equals(solrActionType)) {
-            messageSender.sendIndexingOperation(userid, PIDs.get(targetId), childPids,
-                    IndexingActionType.ADD_SET_TO_PARENT);
-        } else if (UPDATE_DESCRIPTION.equals(solrActionType)) {
-            messageSender.sendIndexingOperation(userid, PIDs.get(targetId), childPids,
-                    IndexingActionType.UPDATE_DESCRIPTION);
-        } else if (PUBLISH.equals(solrActionType)) {
-            for (PID childPid : childPids) {
-                messageSender.sendIndexingOperation(userid, childPid,
-                        IndexingActionType.UPDATE_STATUS);
-            }
-        } else {
+        IndexingActionType indexingActionType;
+        switch (actionType) {
+        case MOVE:
+            indexingActionType = IndexingActionType.MOVE;
+            break;
+        case ADD:
+            indexingActionType = IndexingActionType.ADD_SET_TO_PARENT;
+            break;
+        case REMOVE:
+            indexingActionType = IndexingActionType.DELETE_SOLR_TREE;
+            break;
+        case UPDATE_DESCRIPTION:
+            indexingActionType = IndexingActionType.UPDATE_DESCRIPTION;
+            break;
+        case SET_AS_PRIMARY_OBJECT:
+            indexingActionType = IndexingActionType.RECURSIVE_ADD;
+            break;
+        case EDIT_ACCESS_CONTROL:
```

Comment:
do you intend for this to allow the EDIT_ACCESS_CONTROL and MARK_FOR_DELETION cases to fall through to RESTORE_FROM_DELETION?

Reply:
Sure, i've added a comment